modules/calc/src/main/java/com/opengamma/strata/calc/runner/TaskBuilder.java

```diff
@@ -0,0 +1,204 @@
+/**
+ * Copyright (C) 2016 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.calc.runner;
+
+import java.util.List;
+
+import com.google.common.collect.ArrayListMultimap;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ListMultimap;
+import com.opengamma.strata.basics.CalculationTarget;
+import com.opengamma.strata.calc.CalculationRunner;
+import com.opengamma.strata.calc.marketdata.mapping.MarketDataMappings;
+import com.opengamma.strata.calc.runner.function.CalculationFunction;
+import com.opengamma.strata.collect.ArgChecker;
+
+/**
+ * Builder used to create calculation tasks.
+ * <p>
+ * The {@link CalculationRunner} produces a grid of results, with a row for each target
+ * and a column for each measure. The targets and columns that define the grid of results
+ * are passed in using an instance of {@code CalculationTasks}.
+ * This builder is used to produce the tasks for a single target.
+ * <p>
+ * The result of the builder consists of a number of tasks, each containing one or more task cells.
+ * Each task cell corresponds to a single column in the row.
+ */
+public final class TaskBuilder {
```

Comment:
This exists to allow the functions to intercede in building the tasks. Right now, it doesn't do much, but my intention is for it to grow to handle things like pre-calculating securities shared between bonds, futures and options etc.

Another use would be to allow a different function to be picked for some columns, based on the parameters.

Another use might be to setup certain cells as being unable to calculate for the target type.

(In general, this commit makes the function the hub of all pluggable logic. Part of sensibly allocating effort is controlling the creation of the tasks. Allowing the function to control them outside the control of the framework was fraught with difficulties, especially in ensuring that a result is generated for every cell. As such, I adopted this design - a mutable builder which functions can call.)